rust/processor/src/processors/emojicoin_dot_fun/processor.rs

```diff
@@ -392,14 +487,56 @@ impl ProcessorTrait for EmojicoinProcessor {
-                                ArenaEvent::Exit(exit) => arena_exit_events_db
-                                    .push(ArenaExitEventModel::new(txn_info.clone(), exit)),
-                                ArenaEvent::Swap(swap) => arena_swap_events_db
-                                    .push(ArenaSwapEventModel::new(txn_info.clone(), swap)),
+                                ArenaEvent::Melee(melee) => {
+                                    arena_melee_events_db.push(ArenaMeleeEventModel::new(
+                                        txn_info.clone(),
+                                        melee.clone(),
+                                    ));
+                                },
+                                ArenaEvent::Enter(enter) => {
+                                    arena_position_db
+                                        .push(ArenaPositionDiffModel::from(enter.clone()));
+                                    arena_enter_events_db
+                                        .push(ArenaEnterEventModel::new(txn_info.clone(), enter))
+                                },
+                                ArenaEvent::Exit(exit) => {
+                                    arena_position_db
+                                        .push(ArenaPositionDiffModel::from(exit.clone()));
+                                    arena_exit_events_db
+                                        .push(ArenaExitEventModel::new(txn_info.clone(), exit))
+                                },
+                                ArenaEvent::Swap(swap) => {
+                                    let swaps = (last_swaps.0.unwrap(), last_swaps.1.unwrap());
+                                    // This checks that the two previous swaps do indeed correspond
+                                    // to an arena swap. If stars align, two unrelated swaps (not
+                                    // part of an arena swap) from the same transaction could have
+                                    // net proceeds equal to input amount, and the second swap
+                                    // could have net proceeds equal to the net proceeds of the
+                                    // arena swap, but it is highly unlikely. Moreover, this check
+                                    // is a "just to be sure" check: in theory, last_swaps should
+                                    // always contain the correct swaps due to the way events are
+                                    // emitted.
+                                    if swaps.0.net_proceeds != swaps.1.input_amount
+                                        || (swap.emojicoin_0_proceeds != swaps.1.net_proceeds
+                                            && swap.emojicoin_1_proceeds != swaps.1.net_proceeds)
+                                    {
+                                        bail!("The two previous swaps to an arena swap are not related to the arena swap.");
+                                    }
```

Comment:
Will this result in a processor crash and failure to continue indexing in production?